docker-app/qfieldcloud/core/tests/test_api.py

```diff
@@ -26,3 +44,35 @@ def test_api_status_cache(self):
         toc = time.perf_counter()
 
         self.assertGreater(toc - tic, 0)
+
+    def test_api_pagination_limitoffset(self):
+        """Test LimitOffset pagination custom implementation"""
+        # Authenticate client
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.token.key)
+
+        page_size = 35
+        offset = 36
+        unlimited_count = Project.objects.all().count()
+        self.assertEqual(unlimited_count, 500)
+
+        # Obtain response with LIMIT
+        results_with_pagination = self.client.get(
+            "/api/v1/projects/", {"limit": page_size}
+        ).json()
+        self.assertEqual(len(results_with_pagination), page_size)
+
+        # Obtain response with LIMIT and OFFSET
+        results_with_offset = self.client.get(
+            "/api/v1/projects/", {"limit": page_size, "offset": offset}
+        ).json()
+
+        # Test page size
+        self.assertEqual(len(results_with_offset), page_size)
+
+        # Obtain without pagination (= control test)
+        results_without_pagination = self.client.get(
+            "/api/v1/projects/",
+        ).json()
+
+        # Test length (this is super slow -- 1 minute or so -- because of serialization)
+        self.assertEqual(len(results_without_pagination), unlimited_count)
```

Comment:
If there is no pagination by default, then just decrease the number of projects we have to save some CI time .